### PR TITLE
Add filter to allow VIP SMTP Bypass

### PR DESCRIPTION
**Pull request by mjangda**
By default, all VIP environments are automatically configured to send transactional emails (e.g. password reset, new comment) generated by WordPreess using VIP SMTP servers.

For cases, where a third party provider needs to be used for all emails from the site, including both marketing and transaction emails, the new filter (`vip_smtp_enabled`) can be use to disable loading of the VIP SMTP mailer and loading a custom one.

Note that loading a custom one is important otherwise no emails will get sent.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change works and has been tested locally (or has an appropriate fallback).
- [x] This change works and has been tested on a Go sandbox.
- [x] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).

## Steps to Test

Need to test this on a sandbox, since local falls back to default mail.

1. Apply patch.
1. Trigger password reset and verify you get email.
1. Add `add_filter( 'vip_smtp_enabled', '__return_false' );`.
1. Trigger password reset and verify you do not get an email.

### CODE REVIEW

**Comment by hanifn** [comment]
Tested this on a sandboxed site and seems to be working as expected
Screenshot 2020-05-12 14 40 58
Tests seems to be failing though

File: tests/test-vip-mail.php [hunk 1]
```diff
@@ -6,6 +6,21 @@ public function setUp() {
 		reset_phpmailer_instance();
 	}
 
+	public function test__disable_vip_smtp() {
+		$GLOBALS['all_smtp_servers'] = [ 'server1', 'server2' ];
+
+		add_filter( 'vip_smtp_enabled', '__return_false' );
```
Comment (mjangda) [inline tests/test-vip-mail.php:12]: This runs too late for the test to actually work. Need to work to run before `init` somehow. Or re-initialize the VIP_SMTP class.

Comment (hanifn) [reply tests/test-vip-mail.php:12]: Can we just do another `( new VIP_SMTP() )->init()` after that filter was set?

Comment (hanifn) [reply tests/test-vip-mail.php:12]: Ok tested that out and it didn't work

**Comment by brandon-m-skinner** [comment]
@mjangda Is this still being worked on?

**Comment by mjangda** [comment]
This would be nice, but is not critical. Closing.

**Pull request closed without merging**